### PR TITLE
Wavelet denoising integration

### PR DESCRIPTION
**Pull request by Sand3r-**
The feature ties all pieces related to wavelet denoising up together exposing an API to be used from elsewhere, allowing for signal denoising through Undecimated Wavelet Transform with usage of Daubechies 4 filters.

### CODE REVIEW

**Comment by Sand3r-** [comment]
[Implement wavelet denoising](https://trello.com/c/KzPr7RPc/205-implement-wavelet-denoising)

**Review by Regis128 — DISMISSED** [review]
Nice job, looks good to me

File: src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    WaveletCoefficients& coefficients, size_t signalLength)
+{
+    Signal highFreqCoefficients(signalLength);
+    ExtractCoefficientsForNoiseEstimation(highFreqCoefficients, coefficients);
+    DataType noiseTreshold = noiseEstimator.Estimate(highFreqCoefficients);
+    SoftThresholder tresholder(noiseTreshold);
+    return tresholder(std::move(coefficients));
+}
+
+static inline WaveletCoefficients Decompose(const WaveletDecomposerRef& decomposer,
```
Comment (gmrukwa) [inline src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp]: I don't see a point for creating this and following function, but there's nothing bad about it also, so fine :D

**Review by gmrukwa — DISMISSED** [review]
Nice & clean code. Good job! :)

File: src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp (hunk 1 above)
Comment (Sand3r-) [reply src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp]: Yeah, it looks redundant, but the point was to keep all the code in `Denoise` on the same level of abstraction. This gets inlined anyway, so no performance is damaged.

**Comment by Sand3r-** [comment]
Sorry, I had to rebase. Nothing has really changed since your last approves @Regis128 @gmrukwa.

File: src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp (hunk 1 above)
Comment (gmrukwa) [reply src/Spectre.libWavelet/DaubechiesFiltersDenoiser.cpp]: Yeah, that's what I saw.

**Review by gmrukwa — APPROVED** [review]

**Review by Regis128 — APPROVED** [review]

**Pull request merged**